.github/workflows/ci.yml

```diff
@@ -0,0 +1,30 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches: main
+
+jobs:
+  setup:
```

Comment:
I'm a little confused by this job. It seems to setup yarn then... do nothing?

Reply:
My believe here is that it would cache the setup business for the `lint` and `build` stages. Do you not think that is happening?

Reply:
Oh I see. I do see that happening.

The latest build shows `setup` with _Cache not found for input keys: Linux-node-c8816971176060ebbe1b9ae3e5ffd1a889ad92fd0d3fc66850a943b6244dc7b7, Linux-node-_, then both  `build`/`lint` show _Cache restored successfully_ using that key.

I'd expect good caching from commit to commit without this, so I'm not sure how much this is actually gaining you, but I can confirm it's doing as intended.